mesonbuild/optinterpreter.py

```diff
@@ -146,13 +170,14 @@ def evaluate_statement(self, node: mparser.BaseNode) -> None:
         self.func_option(posargs, kwargs)
 
     @typed_kwargs('option',
-                  KwargInfo('type', str, required=True),
+                  KwargInfo('type', str, required=True,
+                            validator=in_set_validator({'string', 'boolean', 'integer', 'combo', 'array', 'feature'})),
                   KwargInfo('description', str, default=''),
                   KwargInfo('yield', bool, default=coredata.default_yielding, since='0.45.0'),
-                  KwargInfo('choices', (ContainerTypeInfo(list, str), type(None))),
+                  KwargInfo('choices', (ContainerTypeInfo(list, str), NoneType)),
                   KwargInfo('value', object),
```

Comment:
This exists solely to allow the argument through without checking anything -- allow_unknown works well with that. :)